### PR TITLE
Add release docs sections describing the role of a Release Manager

### PR DESCRIPTION
**Pull request by lbdreyer**
### Description
@tkknight, earlier we discussed adding a short section to the releases docs to outline what the release manager does. What do you think of this?

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/src/developers_guide/release.rst [hunk 1]
```diff
@@ -11,6 +11,20 @@ The summary below is of the main areas that constitute the release.  The final
...
+---------------
+A Release Manager will be nominated for each release of Iris. This role involves:
+
+* deciding which features and bug fixes should be included in the release
+* managing the project board for the release
+* updating the iris dev community via the discussion board
+
+The Release Manager will make the release, ensuring that all the steps outlined
+on this page are completed.
+
```
Comment (tkknight) [inline docs/src/developers_guide/release.rst:26]: I would suggest also linking thru to the discussion categories https://www-avd/~itkt/iris-docs-dev/src/_build/html/developers_guide/release.html stating this is where the release manager will document intent and capture any discussion points about a release.

Perhaps also update the Iris 3.2 and 3.3 discussions first post to link to this release guidance for reference.

File: docs/src/developers_guide/release.rst [hunk 2]
```diff
@@ -11,6 +11,20 @@ The summary below is of the main areas that constitute the release.  The final
 section details the :ref:`iris_development_releases_steps` to take.
 
 
+.. _release-manager:
```
Comment (tkknight) [inline docs/src/developers_guide/release.rst:14]: This is just a preference, but I suggest using underscores and not hypens, I think most other doc bookmarks use underscores too.

**Review by tkknight — CHANGES_REQUESTED** [review]

**Comment by lbdreyer** [comment]
Thanks for the review @tkknight . I have now hopefully addressed your review comments

File: docs/src/developers_guide/release.rst [hunk 3]
```diff
@@ -257,6 +258,7 @@ Post Release Steps
 
 .. _SciTools/iris: https://github.com/SciTools/iris
 .. _tag on the SciTools/Iris: https://github.com/SciTools/iris/releases
+.. _GitHub Discussions forum: https://github.com/SciTools/iris/discussions
```
Comment (tkknight) [inline docs/src/developers_guide/release.rst:261]: Suggest we could link directly to the release category https://github.com/SciTools/iris/discussions/categories/releases using the text **GitHub Releases Discussion Forum**

**Review by tkknight — CHANGES_REQUESTED** [review]

**Review by tkknight — APPROVED** [review]

File: docs/src/developers_guide/release.rst [hunk 4]
```diff
@@ -243,6 +258,7 @@ Post Release Steps
 
 .. _SciTools/iris: https://github.com/SciTools/iris
 .. _tag on the SciTools/Iris: https://github.com/SciTools/iris/releases
+.. _GitHub Discussions forum: https://github.com/SciTools/iris/discussions/categories/releases
```
Comment (tkknight) [inline docs/src/developers_guide/release.rst:261]:
```suggestion
.. _GitHub Releases Discussion Forum: https://github.com/SciTools/iris/discussions/categories/releases
```

Comment (lbdreyer) [reply docs/src/developers_guide/release.rst]: Thanks @tkknight  !

**Comment by tkknight** [comment]
The linkcheck has failed due to http://www.ecmwf.int/ being unreachable, the website looks to be up again just this minute.  I have rerun the cirrus task that failed, lets see what happens.  If it fails again I suggest retrying the CI tomorrow.  If the website persists as being offline we can add it to the exclude list https://github.com/SciTools/iris/blob/main/docs/src/conf.py#L303

**Pull request merged**